### PR TITLE
Update go-mod references

### PR DESCRIPTION
**Pull request by thitch97**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
 ```toml
 [[requires]]
 
-  # The name of the Go Mod dependency is "go-mod". This value is
+  # The name of the Go Mod dependency is "go-mod-vendor". This value is
   # considered part of the public API for the buildpack and will not change
   # without a plan for deprecation.
-  name = "go-mod"
+  name = "go-mod-vendor"
```
Comment (ryanmoran) [inline README.md:20]: As the comment above this field mentions, this is part of the public API and will not be changed without a plan for deprecation. Do we have a plan? Or should this remain how it was?

File: rfcs/0000-template.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# {{TITLE: a human-readable title for this RFC!}}
```
Comment (ryanmoran) [inline rfcs/0000-template.md:1]: Are these RFC file supposed to be in this PR?

File: README.md (hunk 1 above)
Comment (sophiewigmore) [reply README.md:20]: We aren't changing this from `go-mod`, the README change was an accidental find and replace, thanks for catching that!

File: mod/mod.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	Dependency = "go-mod"
+	Dependency = "go-mod-vendor"
```
Comment (ryanmoran) [inline mod/mod.go:16]: @sophiewigmore but it does look like it was changed, re: https://github.com/paketo-buildpacks/go-mod-vendor/pull/72/files#r454453176

**Pull request merged**